contracts/EscrowedNori.sol

```diff
@@ -0,0 +1,725 @@
+      memory escrowScheduleDetails = new EscrowScheduleDetail[](
+        accounts.length
+      );
+    for (uint256 i = 0; i < accounts.length; i++) {
+      uint256[] memory escrowScheduleIds = _addressToEscrowScheduleIds[
+        accounts[i]
+      ];
+      for (uint256 j = 0; j < escrowScheduleIds.length; j++) {
+        escrowScheduleDetails[i] = getEscrowSchedule(
+          accounts[i],
+          escrowScheduleIds[j]
+        );
+      }
+    }
+    return escrowScheduleDetails;
+  }
+
+  /**
+   * @notice Returns all governing settings for an escrow schedule.
+   */
+  function getEscrowSchedule(address account, uint256 escrowScheduleId)
+    public
+    view
+    returns (EscrowScheduleDetail memory)
+  {
+    EscrowSchedule storage escrowAcount = _addressToEscrowSchedules[account][
+      escrowScheduleId
+    ];
+    return
+      EscrowScheduleDetail(
+        escrowAcount.currentAmount,
+        account,
+        escrowAcount.startTime,
+        escrowAcount.endTime,
+        escrowAcount.claimedAmount,
+        escrowAcount.totalQuantityRevoked,
+        escrowAcount.exists
+      );
+  }
+
+  /**
+   * @notice Released balance less any claimed amount at current block timestamp.
+   */
+  function claimableBalanceOf(address account) public view returns (uint256) {
+    return _claimableBalanceOf(account, block.timestamp); // solhint-disable-line not-rely-on-time, this is time-dependent
+  }
+
+  /**
+   * @notice Overridden standard ERC777.burn that will always revert
+   *
+   * @dev This function is not currently supported from external callers so we override it so that we can revert.
+   */
+  function burn(uint256, bytes memory) public pure override {
+    revert("eNORI: burning not supported");
+  }
+
+  /**
+   * @notice Overridden standard ERC777.operatorBurn that will always revert
+   *
+   * @dev This function is not currently supported from external callers so we override it so that we can revert.
+   */
+  function operatorBurn(
+    address,
+    uint256,
+    bytes memory,
+    bytes memory
+  ) public pure override {
+    revert("eNORI: burning not supported");
+  }
+
+  /**
+   * @notice Wraps minting of wrapper token and escrow schedule setup.
+   *
+   * @dev If `startTime` is zero no escrow schedule is set up.
+   * Satisfies situations where funding of the account happens over time.
+   *
+   * @param amount uint256 Quantity of `_bridgedPolygonNori` to deposit
+   * @param userData bytes extra information provided by the user (if any)
+   * @param operatorData bytes extra information provided by the operator (if any)
+   */
+  function _depositFor(
+    uint256 removalId,
+    uint256 amount,
+    bytes calldata userData,
+    bytes calldata operatorData
+  ) internal returns (bool) {
+    address recipient = removalId.supplierAddress();
+    uint256 escrowScheduleId = _removal.getEscrowScheduleIdForRemoval(
+      removalId
+    );
+    // TODO error handling/ what to do if this value doesn't exist??
```

Comment:
seems like something would be really messed up if it didnt

Can you describe a case where it doesnt exist?

Reply:
so this is something I want to discuss a bit...  In theory, you can't even mint a removal unless a value for this mapping has been provided at the time of minting, so it should always be here.  However, I am holding onto a vision for a world where our business logic makes more sense and the 10-year insurance period for a removal actually starts in the year of that removal and isn't based on the arbitrary date of the signing of the NRT agreement.  This would be really nice because 1) it's extremely clear and makes a lot of sense and 2) it reduces complexity in our app because we don't have to keep track of this date.  I was thinking that maybe a value of `0` for the escrow schedule id could be interpreted instead to mean that we use the vintage of the removal as the start date for the removal's escrow schedule.  In this way, we could seamlessly transition to a situation where we don't keep track of agreement dates and every removal is just insured on a period spanning 10 years from the actual sequestration.

For now, maybe we just don't check this value (and the worst case scenario is that this value is 0 which means that the escrow schedule will start in 1970 and all of the tokens will be immediately claimable, so at least it doesn't hurt the supplier it only hurts us).  In the future, it would involve minimal code changes to move to a world where each removal uses an automatic start date based on its vintage

Reply:
TL;DR - it would be really messed up if it didn't exist - we want to create escrow schedules when a removal is minted, and a missing value for this should never happen.  if the schedule can't be found I think we revert.